rust/operator-binary/src/controller.rs

```diff
@@ -237,31 +236,30 @@ pub async fn reconcile_opa(opa: Arc<OpaCluster>, ctx: Arc<Ctx>) -> Result<Action
         .map(Cow::Borrowed)
         .unwrap_or_default();
 
+    let vector_aggregator_address = resolve_vector_aggregator_address(&opa, client)
+        .await
+        .context(ResolveVectorAggregatorAddressSnafu)?;
+
     let server_role_service = build_server_role_service(&opa, &resolved_product_image)?;
     // required for discovery config map later
     let server_role_service = cluster_resources
         .add(client, server_role_service)
         .await
         .context(ApplyRoleServiceSnafu)?;
 
-    let (opa_builder_role_serviceaccount, opa_builder_role_rolebinding) =
-        build_opa_builder_serviceaccount(
-            &opa,
-            &resolved_product_image,
-            &ctx.opa_bundle_builder_clusterrole,
-        )?;
-    cluster_resources
-        .add(client, opa_builder_role_serviceaccount)
-        .await
-        .context(ApplyRoleServiceAccountSnafu)?;
-    cluster_resources
-        .add(client, opa_builder_role_rolebinding)
+    let (rbac_sa, rbac_rolebinding) = build_rbac_resources(opa.as_ref(), APP_NAME);
+    client
+        .apply_patch(OPA_CONTROLLER_NAME, &rbac_sa, &rbac_sa)
         .await
-        .context(ApplyRoleRoleBindingSnafu)?;
-
-    let vector_aggregator_address = resolve_vector_aggregator_address(&opa, client)
+        .with_context(|_| ApplyServiceAccountSnafu {
+            name: rbac_sa.name_any(),
+        })?;
+    client
+        .apply_patch(OPA_CONTROLLER_NAME, &rbac_rolebinding, &rbac_rolebinding)
         .await
-        .context(ResolveVectorAggregatorAddressSnafu)?;
+        .with_context(|_| ApplyRoleBindingSnafu {
+            name: rbac_rolebinding.name_any(),
+        })?;
```

Comment:
Was this an explicit decision to leave the SA and Binding out of cluster resources?
If not the `build_rbac_resources` does not set any owner reference and these resources will not be cleaned up?

Reply:
These are now added to the cluster resources.